python/foglamp/common/configuration_manager.py

```diff
@@ -971,6 +975,75 @@ def diff(lst1, lst2):
 
         return result
 
+    async def delete_recursively_parent_category(self, category_name):
```

Comment:
The name of this is misleading, it implies it is deleting a parent of the category, which it is not.